app/views/articles/_v2_form.html.erb

```diff
@@ -37,7 +37,7 @@
 </style>
 
 <main id="main-content"
-  data-article="<%= article.to_json(only: %i[id title cached_tag_list published body_markdown main_image organization_id canonical_url updated_at], methods: %i[series all_series]) %>"
+  data-article="<%= article.to_json(only: %i[id title cached_tag_list published body_markdown main_image organization_id canonical_url updated_at], methods: %i[series all_series decoded_title]) %>"
```

Comment:
I introduced `decoded_title` which only decodes the HTML entities. Sanitization happens with FilterXSS, and also sanitize_and_decode when displaying the article.

Reply:
Refactor (blocking):

The issue is not completely fixed.

If a post is edited, the title is correct in edit/preview mode.

![image](https://user-images.githubusercontent.com/833231/124784153-a9bc6180-df13-11eb-919d-69001540a092.png)

![image](https://user-images.githubusercontent.com/833231/124784192-b2ad3300-df13-11eb-9b03-87ae09737445.png)

If the post is saved as a draft, the script tags are stripped due to XSS vulnerabilities.

![image](https://user-images.githubusercontent.com/833231/124784565-0cadf880-df14-11eb-93c9-6b670926b1ce.png)

Publishing a post is fine.

![image](https://user-images.githubusercontent.com/833231/124784456-f2741a80-df13-11eb-8453-7900f1952291.png)

One other thing to note is the document.title is not decoded when published. In draft mode it's fine though.

![image](https://user-images.githubusercontent.com/833231/124784729-323b0200-df14-11eb-9371-c71375f7ee8b.png)

Reply:
@rhymes, are you OK with the `decoded_title` approach? I wonder if the title returned instead should always be decoded.

Reply:
I'm okay either way! It seems like we use both in the component...